modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/RuleEditorModal.es.js

```diff
@@ -13,135 +13,128 @@
  */
 
 import ClayButton from '@clayui/button';
+import {useResource} from '@clayui/data-provider';
 import {ClayInput} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
 import ClayModal, {useModal} from '@clayui/modal';
-import {RuleEditor} from 'dynamic-data-mapping-form-builder';
-import React, {
-	useCallback,
-	useContext,
-	useEffect,
-	useRef,
-	useState,
-} from 'react';
+import {PagesVisitor} from 'dynamic-data-mapping-form-renderer';
+import {fetch} from 'frontend-js-web';
+import React, {useContext, useMemo, useRef, useState} from 'react';
 
 import AppContext from '../../AppContext.es';
+import {ADD_DATA_LAYOUT_RULE, UPDATE_DATA_LAYOUT_RULE} from '../../actions.es';
 import DataLayoutBuilderContext from '../../data-layout-builder/DataLayoutBuilderContext.es';
-import {getItem} from '../../utils/client.es';
 import ModalWithEventPrevented from '../modal/ModalWithEventPrevented.es';
+import {Editor} from '../rule-builder/editor/Editor.es';
 
-class RuleEditorWrapper extends RuleEditor {
-	getChildContext() {
-		return {
-			store: {
-				editingLanguageId: Liferay.ThemeDisplay.getDefaultLanguageId(),
-			},
-		};
-	}
+const INITIAL_RULE = {
```

Comment:
I'm not sure if this can be placed here or on the Editor component. @matuzalemsteles